### PR TITLE
[receiver/cloudfoundry] Enable component

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** 
The Cloud Foundry receiver has already been implemented and merged into the contrib repository, however, testing was incomplete and the receiver wasn't enabled. This change adds the required unit testing and enables the receiver.

**Link to tracking Issue:**
#5320

**Testing:** 
This change is tests, and additional testing was done by running a Tanzu application service.

**Documentation:**
No documentation was added as this is a testing change.

### CODE REVIEW

File: receiver/cloudfoundryreceiver/receiver_test.go [hunk 1]
```diff
@@ -14,4 +14,52 @@
...
+	cfg := factory.CreateDefaultConfig().(*Config)
+	params := componenttest.NewNopReceiverCreateSettings()
+
+	receiver, err := newCloudFoundryReceiver(
+		params,
+		*cfg,
+		nil,
+	)
+
+	require.Error(t, err)
```
Comment (rmfitzpatrick) [inline receiver/cloudfoundryreceiver/receiver_test.go:63]: could be helpful to have an `EqualError()` or similar to provide more info about the failure expectation

File: receiver/cloudfoundryreceiver/stream_test.go [hunk 2]
```diff
@@ -14,4 +14,100 @@
...
+		cfg.RLPGateway.HTTPClientSettings,
+		componenttest.NewNopHost())
+
+	innerCtx, _ = context.WithCancel(context.Background())
+
+	envelopeStream, createErr = streamFactory.CreateStream(
+		innerCtx,
+		INVALID_SHARD_ID)
+
+	require.Error(t, createErr)
```
Comment (rmfitzpatrick) [inline receiver/cloudfoundryreceiver/stream_test.go:111]: same error content request

File: receiver/cloudfoundryreceiver/uaa_test.go [hunk 3]
```diff
@@ -14,4 +14,51 @@
...
+
+	cfg.UAA.LimitedHTTPClientSettings.Endpoint = ""
+
+	uaa, err := newUAATokenProvider(
+		zap.NewNop(),
+		cfg.UAA.LimitedHTTPClientSettings,
+		cfg.UAA.Username,
+		cfg.UAA.Password)
+
+	require.Error(t, err)
```
Comment (rmfitzpatrick) [inline receiver/cloudfoundryreceiver/uaa_test.go:62]: same

**Review by rmfitzpatrick — APPROVED** [review]

**Review by mstumpfx — APPROVED** [review]

File: receiver/cloudfoundryreceiver/receiver_test.go [hunk 4]
```diff
@@ -14,4 +14,52 @@
...
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+)
+
+// Test to make sure a new receiver can be created properly, started and shutdown with the default config
+func TestDefaultValidReceiver(t *testing.T) {
```
Comment (jpkrohling) [inline receiver/cloudfoundryreceiver/receiver_test.go:27]: This should just be another test in this table test: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/013177e592b724dc217dc9ad958242db76158f4e/internal/components/receivers_test.go#L43

**Comment by jpkrohling** [comment]
Please, also add an entry to the changelog.

**Review by jpkrohling — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**